grails-app/controllers/org/broadinstitute/orsp/CommentsController.groovy

```diff
@@ -1,13 +1,57 @@
 package org.broadinstitute.orsp
 
+import grails.converters.JSON
+import grails.rest.Resource
+import groovy.util.logging.Slf4j
+
+@Slf4j
+@Resource(readOnly = false, formats = ['JSON', 'APPLICATION-MULTIPART'])
 class CommentsController extends AuthenticatedController {
 
+    CommentsService commentsService
+
     def index = { list }
 
+    @Deprecated
     def list = {
         Issue issue = queryService.findByKey(params.id)
         Collection<Comment> comments = Comment.findAllByProjectKey(params.id)
         [issue: issue, comments: comments]
     }
 
+    def saveNewComment() {
+        try {
+            Comment savedComment = commentsService.addComment(params.id, params.comment)
+            response.status = 200
+            render savedComment as JSON
+        } catch (IllegalArgumentException e) {
+            response.status = 400
+            render([error: e.message] as JSON)
+        } catch (Exception e) {
+            response.status = 500
+            render([error: e.message] as JSON)
+        }
+    }
+
+    def getComments() {
+        JSON.registerObjectMarshaller(Comment) {
+            def output = [:]
```

Comment:
Two minor things here. 
1. This is actually a `LinkedHashMap<Object, Object>`, avoid def whenever possible. 
2. I see this pattern in UserController too, which is handy - but we're spreading out these marshallers and there's the potential for DRY violations this way. I'm wondering if these should be pulled up into a utility class (i.e. `UtilityClass.marshallObjects()`, `UtilityClass.marshallComment()` or something similar) so they can be more easily referenced from anywhere.

Reply:
1- Def value is now refactored.
2- This makes sense, as you suggested I moved marshallers to a Utility class.